File: Searching Algorithms/Binary_search_recursively.cpp
#include <bits/stdc++.h>
using namespace std;

int binarySearch(int arr[], int n, int left, int right, int target)
{
        if(left>right)
        return -1;
    
        int mid = left + (right-left)/2;

        if(arr[mid]==target)
        return mid;
        else if(arr[mid]>target)
        binarySearch(arr, n, left, mid-1, target);
        else
        binarySearch(arr, n, mid+1, right, target);
    
    
    
}
int main(){
    int n;
    cin >> n;

    int arr[n];
    for(int i=0; i<n; i++)
    cin >> arr[i];

    int target;
    cin >> target;

    int find = binarySearch(arr, n, 0, n-1, target);
    cout << find << endl;

return 0;
}

File: Searching Algorithms/BinarySearchRecursively .py
def binary_search(start,end,test_list,target): 
  if start<=end:
     mid = (start+end) // 2
     if test_list[mid] == target:
       return mid +1
     elif target < test_list[mid]:
       return binary_search(start,mid-1,test_list,target)
     elif target > test_list[mid]:
       return binary_search(mid+1,end,test_list,target)
  else:
     return -1

test_list = [ 45, 78, 56, 23, 89 ,965]
target = 34
location = binary_search(0, len(test_list)-1,test_list, target)

if location != -1:
	print("Element is present at ", str(location))
else:
	print("Element is not present in test array.")
